code/_onclick/click.dm

```diff
@@ -96,7 +96,7 @@
 
 		return
 
-	if(!isturf(loc)) // This is going to stop you from telekinesing from inside a closet, but I don't shed many tears for that
+	if(!isturf(loc) && !find_holder_of_type(A, src)) // Can't touch anything from inside a locker/sleeper etc, unless it's inside our inventory.
```

Comment:
I'm looking through unsorted.dm and you will not believe the amount of duplicate procs we have
There's also no is_holder_of which is shameful